pkg/controller/s3/bucket/CORSConfig.go

```diff
@@ -131,3 +143,36 @@ func (in *CORSConfigurationClient) Delete(ctx context.Context, bucket *v1beta1.B
 	).Send(ctx)
 	return awsclient.Wrap(err, corsDeleteFailed)
 }
+
+// LateInitialize does nothing because CORSConfiguration might have been deleted
+// by the user.
+func (in *CORSConfigurationClient) LateInitialize(ctx context.Context, bucket *v1beta1.Bucket) error {
+	external, err := in.client.GetBucketCorsRequest(&awss3.GetBucketCorsInput{Bucket: awsclient.String(meta.GetExternalName(bucket))}).Send(ctx)
+	if err != nil {
+		// Short stop method for requests in a region without CORS Support
+		if resource.Ignore(s3.CORSConfigurationNotFound, err) != nil {
+			return awsclient.Wrap(err, corsGetFailed)
+		}
+		return nil
+	}
+
+	// We need the second check here because by default the CORS is not set
+	if external.GetBucketCorsOutput == nil || len(external.CORSRules) == 0 {
+		return nil
+	}
+
+	in.logger.Debug(fmt.Sprintf("called LateInitialize for %s", reflect.TypeOf(in).Elem().Name()), "external output", external.GetBucketCorsOutput)
+
+	if bucket.Spec.ForProvider.CORSConfiguration == nil {
+		bucket.Spec.ForProvider.CORSConfiguration = &v1beta1.CORSConfiguration{}
+	}
+
+	bucket.Spec.ForProvider.CORSConfiguration.CORSRules = GenerateCORSRule(external.CORSRules)
```

Comment:
We can assign the observed state only if `len(bucket.Spec.ForProvider.CORSConfiguration.CORSRules) == 0` so that we don't override user's desired state.